src/main/java/pl/solvd/carina/Navigation.java

```diff
@@ -0,0 +1,114 @@
+    @FindBy(xpath = "//nav[@aria-label='Navigation']//following-sibling::nav")
+    private List<ExtendedWebElement> navigationComponents;
+    @FindBy(xpath = "//nav[@aria-label='Navigation']/ul[@class='md-nav__list']")
+    private List<ExtendedWebElement> navigationList;
+    @FindBy(xpath = "//li[@class='md-nav__item']")
+    private List <ExtendedWebElement> listItem;
+    @FindBy(xpath = "//a[contains(@class,'link--active')]/..")
+    private ExtendedWebElement currentPage;
+    @FindBy(xpath = "//a/../input[contains(@id,'__toc')]/..")
+    private ExtendedWebElement highlitedPage;
+    @FindBy(xpath = "//li[contains(@class, 'nested')]//nav//ul[@class='md-nav__list']")
+    private NestedElement nestedElement;
+    @FindBy(xpath = "//li//following-sibling::*[@class='md-nav__link']")
+    private List<ExtendedWebElement> navigateLink;
+    @FindBy(xpath = "//li//following-sibling::*[@class='md-nav__link'][@href]")
+    private List<ExtendedWebElement> navigateLinkWithoutParent;
+
+    public Navigation(WebDriver driver, SearchContext searchContext) {
+        super(driver, searchContext);
+    }
+
+    public NestedElement getNestedElement() {
+        return nestedElement;
+    }
+
+    public boolean isCarinaFirst() {
+        return navigationComponents.get(0).getText().equals(findExtendedWebElement(By.xpath("//label/a[@title='Carina']")).getText());
+    }
+
+    public boolean isNavigationListPresent() {
+        for (ExtendedWebElement element : navigationList) {
+            if (element.isElementPresent()) {
+            } else {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public boolean isCurrentPageHighlighted() {
+        return highlitedPage.getElement().equals(currentPage.getElement());
+    }
+
+    public boolean isNestedElement() {
+        for (ExtendedWebElement e : navigationList) {
+            if (e.getElement().isSelected() == nestedElement.getRootExtendedElement().getElement().isSelected()) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+        return false;
+    }
+
+    public boolean isHiddenElement() {
+        for (ExtendedWebElement e : listItem) {
+            if (!e.isVisible() || nestedElement.getNestedBlock().contains(e)) {
+                return false;
+            }
+        } return true;
+    }
+
+    public boolean isSubPagesReveals() {
+        for (int i = 0; i < navigationList.size(); ) {
+            if (navigationList.get(i).getElement().isSelected() == nestedElement.getParentItem().getElement().isSelected()) {
+                nestedElement.getParentItem().click();
+                if (nestedElement.getNestedBlock().get(0).isElementPresent(2)) {
+                    return true;
+                } else {
+                    i++;
+                }
+            }
+        }
+        return false;
+    }
+
+    public boolean isRedirectToURL() {
```

Comment:
not all menu items contain links, but all need to be clicked. I tried, but in the end I didn’t use foreach, because I had to step over menu items that did not contain links, so the code turned out to be not so readable ... (